Icebnb/Icebnb/SearchViewController.swift

```diff
@@ -0,0 +1,100 @@
+  
+  func configureNavbar() {
+    filterBtn = UIButton(frame: CGRect(x: 0, y: 0, width: 50, height: 30))
+    filterBtn.setTitle("Filters", for: .normal)
+    filterBtn.setTitleColor(.black, for: .normal)
+    filterBtn.addTarget(self, action: #selector(filterPressed), for: .touchUpInside)
+    navigationItem.rightBarButtonItem = UIBarButtonItem(customView: filterBtn)
+  }
+  
+  func configureSearchController() {
+    searchBarView.placeholder = "Search items"
+    searchBarView.sizeToFit()
+    
+    searchBarView.barTintColor = ColorConstants.barBackgroundColor
+    searchBarView.isTranslucent = true
+    searchBarView.layer.cornerRadius = 1.0
+    searchBarView.clipsToBounds = true
+  }
+  
+  func configureCollection() {
+    collectionView.register(UINib.init(nibName: "PlaceCell", bundle: nil), forCellWithReuseIdentifier: "placeCellIdentifier")
+    collectionView.delegate = self
+    collectionView.backgroundColor = ColorConstants.tableColor
+    
+    let blurEffect = UIBlurEffect(style: UIBlurEffectStyle.extraLight)
+    let blurEffectView = UIVisualEffectView(effect: blurEffect)
+    blurEffectView.frame = collectionView.bounds
+    blurView.backgroundColor = ColorConstants.tableColor
+    blurView.insertSubview(blurEffectView, at: 0)
+  }
+  
+  func collectionView(_ collectionView: UICollectionView,
+                      cellForItemAt indexPath: IndexPath,
+                      containing hit: [String: Any]) -> UICollectionViewCell {
+    let collectionViewCell = collectionView.dequeueReusableCell(withReuseIdentifier: "placeCellIdentifier", for: indexPath) as! PlaceCell
+    let place = Place(json: hit)
+    collectionViewCell.nameLabel.text = place.name
+    collectionViewCell.descriptionLabel.text = (place.price ?? "") + " - " + (place.roomType ?? "" )
+    collectionViewCell.imageView.image = nil
+    if place.thumbnail != nil, let thumbURL = URL(string: place.thumbnail!) {
+      Nuke.loadImage(with: thumbURL, into: collectionViewCell.imageView)
+    }
+    return collectionViewCell
+  }
+  
+  func filterPressed() {
+    let controller = FilterViewController(nibName: "FilterViewController",
+                                          bundle: nil,
+                                          filters: ["Room type" : { ctrl in
+                                            let roomController = TypeFacetViewController(nibName: "TypeFacetViewController", bundle: nil)
+                                            ctrl.navigationController?.pushViewController(roomController, animated: true)
+                                            },
+                                                    "Price": { ctrl in
+                                                      let rangeController = PriceRangeViewController(nibName: "PriceRangeViewController", bundle: nil)
+                                                      ctrl.navigationController?.pushViewController(rangeController, animated: true)
+                                            }])
```

Comment:
A bit hard to read, too long. Would be good to break this down

Reply:
You're right, changed it